libs/vkd3d/device.c

```diff
@@ -528,7 +528,7 @@ static HRESULT vkd3d_instance_init(struct vkd3d_instance *instance,
     vkd3d_free(user_extension_supported);
 
     vr = vk_global_procs->vkCreateInstance(&instance_info, NULL, &vk_instance);
-    vkd3d_free(extensions);
+    vkd3d_free((char **)extensions);
```

Comment:
What's the problem here exactly, const-ness?

Reply:
const, yes. I don't like removing const warnings, since those can be quite serious.

Reply:
We could also make vkd3d_free take `const void*`, but that feels a bit icky too.

Reply:
I think you can eliminate this with `char *const *extensions;` or something similar.

The const is the thing in the wrong place.

Reply:
The const is in the right place I think, because ppEnabledExtensions is `const char * const *extensions`, and you would get a similar warning if you were to assign `char * const` to `const char *`.